src/sds_data_model/_dataframe.py

```diff
@@ -10,6 +10,12 @@
 from rasterio.features import geometry_mask
 from xarray import DataArray
 
+from shapely.wkt import loads 
+from bng_indexer import calculate_bng_index, wkt_from_bng
```

Comment:
You need to add `bng_indexer` to the library dependencies by running:

` poetry add git+https://github.com/Defra-Data-Science-Centre-of-Excellence/bng-indexer.git`

Also, you don't use `calculate_bng_index` in this file:

```suggestion
from bng_indexer import wkt_from_bng
```